rust/lancedb/src/remote/db.rs

```diff
@@ -22,9 +22,16 @@ use crate::table::BaseTable;
 
 use super::client::{ClientConfig, HttpSend, RequestResultExt, RestfulLanceDbClient, Sender};
 use super::table::RemoteTable;
-use super::util::batches_to_ipc_bytes;
+use super::util::{batches_to_ipc_bytes, parse_server_version};
 use super::ARROW_STREAM_CONTENT_TYPE;
 
+// the versions of the server that we support
+// for any new feature that we need to change the SDK behavior, we should bump the server version,
+// and add a feature flag here, so use `self.server_version >= FEATURE_VERSION` to check if the server
+// supports the feature.
+pub const DEFAULT_SERVER_VERSION: semver::Version = semver::Version::new(0, 1, 0);
+pub const MULTIVECTOR_VERSION: semver::Version = semver::Version::new(0, 2, 0);
```

Comment:
I think these belong alongside the `parse_server_version`. You might also consider consolidating into an easy to read struct.

```rust
struct ServerVersion(pub semver::Version);

impl ServerVersion {
    pub fn supports_multivector_endpoint() -> bool { ... }
}
```